### DISCUSSION TITLE
Is CSS Border behavior different with browser?

### DISCUSSION THREAD

**Discussion by kitepad** [c-smile/sciter-js-sdk#157] [Q&A]
When I trying migrate some web app to sciter.js, I found many border shape css code can not work with sciter.js because of border behavior is not same with browser when width/height equals zero. For example,  customize border's ::before/::after can get arrow/triangle in browser, but sciter.js is not. Is it design consideration or bug?

**Comment by c-smile** [comment]
> Is it design consideration or bug?

It is by design. Sciter offers arbitrary shapes by using [vector images](https://sciter.com/lightweight-inline-vector-images-in-sciter/).

So:

```html
<html>
    <head>
        <title>Test</title>
        <style>

@media { /* conventional browser */   

    .triangle-up {
      width: 0;
      height: 0;
      border-left: 50px solid transparent;
      border-right: 50px solid transparent;
      border-bottom: 100px solid red;
    }

}    

@media sciter { /* guess what? */

    .triangle-up {
      width: 100px;
      height: 100px;
      fill:red; stroke:none;
      background: url(path: M 0 100 L 50 0 L 100 100 Z);
      background-size:100%;
    }

}    
        </style>
    </head>
    <body>
 
      <div class="triangle-up"></div>

    </body>
</html>
```

**Comment by c-smile** [comment]
Also making non-rectangular shapes by hacking borders in browsers is an UB (undefined behavior).
Browsers (UAs) are not obligated to join borders in particular way.

For example 10 years ago at W3C we were discussed different options for that:
https://lists.w3.org/Archives/Public/www-style/2010Jan/0536.html

So we may have gradients as joiners instead:

![border joiners](https://www.terrainformatica.com/w3/round-corners-sciter-b.png)

**Reply by kitepad** [reply]
Understand. Thanks for your advice.
